src/handlers/graphical.rs

```diff
@@ -718,31 +718,33 @@ impl GraphicalReportHandler {
                 let byte_start = hl.offset();
                 let byte_end = hl.offset() + hl.len();
                 let start = self.visual_offset(line, byte_start, true).max(highest);
-                let end = self.visual_offset(line, byte_end, false).max(start + 1);
+                let end = if hl.len() == 0 {
+                    start + 1
+                } else {
+                    self.visual_offset(line, byte_end, false).max(start + 1)
+                };
 
                 let vbar_offset = (start + end) / 2;
                 let num_left = vbar_offset - start;
                 let num_right = end - vbar_offset - 1;
-                if start < end {
-                    underlines.push_str(
-                        &format!(
-                            "{:width$}{}{}{}",
-                            "",
-                            chars.underline.to_string().repeat(num_left),
-                            if hl.len() == 0 {
-                                chars.uarrow
-                            } else if hl.label().is_some() {
-                                chars.underbar
-                            } else {
-                                chars.underline
-                            },
-                            chars.underline.to_string().repeat(num_right),
-                            width = start.saturating_sub(highest),
-                        )
-                        .style(hl.style)
-                        .to_string(),
-                    );
-                }
+                underlines.push_str(
+                    &format!(
+                        "{:width$}{}{}{}",
```

Comment:
I just fixed a bug in #316 that also uses these `:width$` format specifiers. It seems that they're not such a good idea when working with ansi. I do see that you apply style after you do that here though, so it might work fine here actually. I just noticed it in this diff as well and maybe this should be checked everywhere :thinking:

Reply:
I believe it's fine in this case, since the actual format arg is `""`. `format!("{:width$}", "", width = x)` should just expand to `x` space chars. I might be misunderstanding what the problem is, but I don't _think_ this is the same bug as the one you found in #316.

Reply:
ohh yes that is correct, I do that too somewhere I think. You're right, this works :D